.github/workflows/ci.yml

```diff
@@ -19,11 +19,10 @@ jobs:
       - uses: actions/checkout@v4
       - name: Set up Python
         uses: actions/setup-python@v5
-        with:
-          python-version: "3.13"
-          cache: "pip"
+      - name: Install uv
+        uses: astral-sh/setup-uv@v6
       - name: Install dependencies
-        run: python -m pip install --upgrade pip pre-commit
+        run: uv pip install --upgrade pip pre-commit
```

Comment:
to do: edit this entire file to be compatible with uv.. https://docs.astral.sh/uv/guides/integration/github/